dss/storage/index_document.py

```diff
@@ -0,0 +1,350 @@
+                    'type': ESDocType.subscription.name,
+                    'values': [subscription_id]
+                }
+            }
+        }
+        response = ElasticsearchClient.get(self.logger).search(
+            index=Config.get_es_index_name(ESIndexType.subscriptions, self.replica),
+            body=subscription_query)
+        hits = response['hits']['hits']
+        assert len(hits) == 1
+        hit = hits[0]
+        assert hit['_id'] == subscription_id
+        subscription = hit['_source']
+        assert 'id' not in subscription
+        subscription['id'] = subscription_id
+        return subscription
+
+    def notify_subscriber(self, subscription: dict):
+        subscription_id = subscription['id']
+        # FIXME: (hannes) brittle coupling with regex in is_bundle_to_index
+        bundle_uuid, _, bundle_version = self.bundle_id.partition(".")
+        transaction_id = str(uuid.uuid4())
+        payload = {
+            "transaction_id": transaction_id,
+            "subscription_id": subscription_id,
+            "es_query": subscription['es_query'],
+            "match": {
+                "bundle_uuid": bundle_uuid,
+                "bundle_version": bundle_version
+            }
+        }
+        callback_url = subscription['callback_url']
+
+        # FIXME wrap all errors in this block with an exception handler
+        if DeploymentStage.IS_PROD():
+            allowed_schemes = {'https'}
+        else:
+            allowed_schemes = {'https', 'http'}
+
+        assert urlparse(callback_url).scheme in allowed_schemes, "Unexpected scheme for callback URL"
+
+        if DeploymentStage.IS_PROD():
+            hostname = urlparse(callback_url).hostname
+            for family, socktype, proto, canonname, sockaddr in socket.getaddrinfo(hostname, port=None):
+                msg = "Callback hostname resolves to forbidden network"
+                assert ipaddress.ip_address(sockaddr[0]).is_global, msg  # type: ignore
+
+        auth = None
+        if "hmac_secret_key" in subscription:
+            auth = HTTPSignatureAuth(key=subscription['hmac_secret_key'].encode(),
+                                     key_id=subscription.get("hmac_key_id", "hca-dss:" + subscription_id))
+        response = requests.post(callback_url, json=payload, auth=auth)
+
+        # TODO (mbaumann) Add webhook retry logic
+        if 200 <= response.status_code < 300:
+            self.logger.info(f"Successfully notified for subscription {subscription_id}"
+                             f" for bundle {self.bundle_id} with transaction id {transaction_id} "
+                             f"Code: {response.status_code}")
+        else:
+            self.logger.warning(f"Failed notification for subscription {subscription_id}"
+                                f" for bundle {self.bundle_id} with transaction id {transaction_id} "
+                                f"Code: {response.status_code}")
+
+
+class BundleTombstoneDocument(IndexDocument):
```

Comment:
As much as I would like to merge this quickly, but this should not be introduced in this PR.

Reply:
My mistake.